lib/Test/Nginx/Util.pm

```diff
@@ -2364,6 +2367,28 @@ retry:
             #warn "pid file not found";
         }
     }
+
+    if (defined $block->shutdown_error_log) {
```

Comment:
I still believe this is a wrong place of doing the shutdown error log check. The check should be performed before the `start_nginx` code label (around `lib/Test/Nginx/Util.pm`, line 1550), right after shutting down the nginx before there. And we also need to do the check in the `END {}` block, around `lib/Test/Nginx/Util.pm`, line 2383. You can add a Perl sub to avoid code duplication between those 2 places.

We can introduce a global variable called `$PrevBlock` to hold the previous test block. This way, we do not need to change the signature of the `run_test()` function at all.